metrics-operator/controllers/common/providers/dynatrace/client/client.go

```diff
@@ -56,21 +56,21 @@ func NewAPIClient(config apiConfig, options ...APIClientOption) *apiClient {
 }
 
 // Do sends and API request to the Dynatrace API and returns its result as a string containing the raw response payload
-func (client *apiClient) Do(ctx context.Context, path, method string, payload []byte) ([]byte, error) {
+func (client *apiClient) Do(ctx context.Context, path, method string, payload []byte) ([]byte, int, error) {
 	if err := client.auth(ctx); err != nil {
-		return nil, err
+		return nil, 0, err
```

Comment:
To keep consistent with the expectation of receiving a http status code from this functions, I suggest to use `http.StatusInternalServerError` for the error cases (see the suggestions below)

```suggestion
		return nil, http.StatusInternalServerError, err
```